File: diagram/generate-pages.js
#!/usr/bin/env node

const fs = require('fs');
const path = require('path');

// Read the pages configuration
const pagesData = JSON.parse(fs.readFileSync('atavic-pages.json', 'utf8'));

// Create pages directory if it doesn't exist
const pagesDir = 'pages';
if (!fs.existsSync(pagesDir)) {
    fs.mkdirSync(pagesDir, { recursive: true });
}

console.log('Generating HTML pages from template...');

// Generate each page
pagesData.pages.forEach(page => {
    let html = pagesData.template;
    
    // Replace template variables
    html = html.replace(/\{\{TITLE\}\}/g, page.title);
    html = html.replace(/\{\{ELEMENT_ID\}\}/g, page.elementId);
    
    // Write the file
    const filepath = path.join(pagesDir, page.filename);
    fs.writeFileSync(filepath, html, 'utf8');
    
    console.log(`✅ Generated: ${filepath}`);
});

console.log(`\n🎉 Successfully generated ${pagesData.pages.length} HTML pages!`);
console.log('📁 All pages are in the ./pages/ directory');
console.log('🔗 These pages are now clickable in the diagram viewer');

File: diagram/generate-pages.py
#!/usr/bin/env python3

import json
import os

# Read the pages configuration
with open('atavic-pages.json', 'r') as f:
    pages_data = json.load(f)

# Create pages directory if it doesn't exist
pages_dir = 'pages'
if not os.path.exists(pages_dir):
    os.makedirs(pages_dir)

print('Generating HTML pages from template...')

# Generate each page
for page in pages_data['pages']:
    html = pages_data['template']
    
    # Replace template variables
    html = html.replace('{{TITLE}}', page['title'])
    html = html.replace('{{ELEMENT_ID}}', page['elementId'])
    
    # Write the file
    filepath = os.path.join(pages_dir, page['filename'])
    with open(filepath, 'w') as f:
        f.write(html)
    
    print(f"✅ Generated: {filepath}")

print(f"\n🎉 Successfully generated {len(pages_data['pages'])} HTML pages!")
print("📁 All pages are in the ./pages/ directory")
print("🔗 These pages are now clickable in the diagram viewer")
